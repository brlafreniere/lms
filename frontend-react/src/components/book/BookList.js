import Axios from "axios";
import React from "react";

import BookTile from "./BookTile";

export default class BookList extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            books: []
        }
    }

    componentDidMount = () => {
        this.queryBooks();
    }

    queryBooks = () => {
        Axios.get(`${process.env.REACT_APP_API_URL}/books`).then(response => {
            this.setState({books: response.data})
        }).catch(console.log)
    }

    render() {
        return (
            <div className='book-group'>
                {this.state.books.map(book => 
                    <BookTile key={book.id} book={book} />
                )}
            </div>
        )
    }
}