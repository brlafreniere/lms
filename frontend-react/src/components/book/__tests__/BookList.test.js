import React from "react";
import Axios from "axios";
import {shallow, mount, render} from "enzyme";

import BookList from "../BookList";

import renderer from "react-test-renderer";
import { MemoryRouter } from "react-router-dom";


test("<BookList /> snapshot test", () => {
    const response = {
        data: [
            {
                id: 1,
                title: "foo"
            },
            {
                id: 2,
                title: "bar"
            }
        ]
    }
    Axios.get = jest.fn()
    Axios.get.mockResolvedValue(response);

    const component = renderer.create(
        <MemoryRouter>
            <BookList />
        </MemoryRouter>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

describe('<BookList />', () => {
    it("queries books on mount", () => {
        //const wrapper = shallow(<BookList />)
    })
})