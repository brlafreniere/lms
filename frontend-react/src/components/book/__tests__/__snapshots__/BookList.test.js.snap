// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookGrid /> snapshot test 1`] = `
<div
  className="book-group"
/>
`;
